grammar.js

```diff
@@ -72,17 +73,29 @@ module.exports = grammar({
         // Pragma
         pragma_directive: $ => seq(
             "pragma",
-            "solidity",
-            repeat(field("version_constraint", $._pragma_version_constraint)),
+            choice($.solidity_pragma_token, $.any_pragma_token),
             $._semicolon,
         ),
 
+        solidity_pragma_token: $ => prec(10, seq(
+            $._solidity,
+            repeat(field("version_constraint", $._pragma_version_constraint)),
+        )),
+
+        any_pragma_token: $ => seq(
+            $.identifier,
+            $.pragma_value,
+        ),
+
+        _solidity: $ => prec(1, "solidity"),
+        pragma_value: $ =>  prec(0, /[^;]+/),
+
         _pragma_version_constraint: $ => seq(
```

Comment:
I was looking at the last test case in this pr (that's changed). 

I'm thinking it might be a good idea to make `pragma_version_constraint` non-hidden, that way it's more obvious for downstream tools that there is this comparison operator + version tuple.